cirq/ops/common_gates.py

```diff
@@ -322,6 +332,29 @@ def _act_on_(self, args: Any):
                                                         tableau.xs[:, q].copy())
             return True
 
+        if isinstance(args, clifford.ActOnStabilizerCHFormArgs):
+            if protocols.is_parameterized(self) or self.exponent % 0.5 != 0:
+                return NotImplemented
+            effective_exponent = self._exponent % 2
+            state = args.state
+            if effective_exponent == 0.5:
+                assert all(
+                    gate._act_on_(args)  # type: ignore
+                    for gate in [ZPowGate(), HPowGate()])
+                state.omega *= (1 + 1j) / (2**0.5)  # type: ignore
+            elif effective_exponent == 1:
+                assert all(
+                    gate._act_on_(args) for gate in  # type: ignore
+                    [ZPowGate(), HPowGate(),
+                     ZPowGate(), HPowGate()])
+                state.omega *= 1j  # type: ignore
+            elif effective_exponent == 1.5:
+                assert all(
+                    gate._act_on_(args)  # type: ignore
+                    for gate in [HPowGate(), ZPowGate()])
+                state.omega *= (1 - 1j) / (2**0.5)  # type: ignore
```

Comment:
> Left a bunch of comments!
> Also, I'm a bit worried about the fact that phases just go unnoticed...
> In `assert_all_implemented_act_on_effects_match_unitary` shouldn't we use
> 
> ```
>       np.testing.assert_allclose(np.reshape(
>             stabilizer_ch_form.state_vector(), protocols.qid_shape(qubits)),
>                                                    state_vector,
>                                                    atol=1e-07, err_msg="inconsistent stabilizer ch form")
> ```
> 
> instead of
> 
> ```
>         testing.assert_allclose_up_to_global_phase(np.reshape(
>             stabilizer_ch_form.state_vector(), protocols.qid_shape(qubits)),
>                                                    state_vector,
>                                                    atol=1e-07)
> ```
> 
> ?
> One of the benefits of the CH-form is that it is phase sensitive. Why are we checking unitary consistency in a phase insensitive way?

It seems that all the gate implementations for some reason are off by a factor of 1j. I.e if I test with `np.testing.assert_allclose` and `state_vector * 1j` then they all succeed...

Reply:
For the record: I was wrong with the -1 phase for the YPowGate - it seems to be fine. I made a mistake `u(Circuit(H(a), Z(a))` should have been instead of `u(Circuit(H(a), Z(a))`

Reply:
I could not find any issue with the implementation of the gates or off-by `1j` factor issue. Even after taking phases into account, only the tests with `global_shift` being introduced by `assert_eigengate_implements_consistent_protocols` were failing. Added additional accounting based on that to `common_gates` instances.

Also, `decompose_rotation` in `SingleQubitCliffordGate` was messing up the global phase. Added a (slightly expensive) post-correction there.

Reply:
Good find! That was the culprit that I saw! :)